server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/DeletedMessagesVaultTest.java

```diff
@@ -602,6 +609,67 @@ public void vaultExportShouldResponseIdempotentSideEffect() throws Exception {
         }
     }
 
+    @Test
+    public void vaultPurgeShouldDeleteExpiredMessageThenExportWithNoEntry() throws Exception {
+        Instant now = Instant.now();
+        when(clock.instant())
+            .thenReturn(now, now.plus(366, ChronoUnit.DAYS));
+
+        bartSendMessageToHomer();
+        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 1);
+
+        homerDeletesMessages(listMessageIdsForAccount(homerAccessToken));
+        WAIT_TWO_MINUTES.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 0);
+
+        purgeVault();
+
+        String fileLocation = exportAndGetFileLocationFromLastMail(EXPORT_ALL_HOMER_MESSAGES_TO_BART, bartAccessToken);
+        try (ZipAssert zipAssert = assertThatZip(new FileInputStream(fileLocation))) {
+            zipAssert.hasNoEntry();
+        }
+    }
+
+    @Test
+    public void vaultPurgeShouldNotDeleteNotExpiredMessageThenExportWithAnEntry() throws Exception {
```

Comment:
`vaultPurgeShouldMakeExportProduceAZipWithAllNonExpiredMessages` take it if you think it's better